File: core_java/src/main/java/com/meta/cj/day1/EmiCalculator.java
package com.meta.cj.day1;

import java.util.Scanner;
public class EmiCalculator{
	
		public static void main(String[] args){
				
				Scanner sc = new Scanner(System.in);
				System.out.println("Enter the amount :");
				double p = sc.nextDouble();
				System.out.println("Enter the interest rate (%) :");
				double r = sc.nextDouble();
				System.out.println("Enter the number of months :");
				double n = sc.nextDouble();
				
				double rn = r / 12 / 100;				
				double emi = p * rn * ( Math.pow(1+rn,n) / (Math.pow(1+rn,n)-1));
				double totalAmount = emi * n;
			    double totalInterest = totalAmount - p;
				
				System.out.println("EMI Amount     :"+Math.round(emi));
				System.out.println("Total Interest :"+Math.round(totalInterest));
				System.out.println("Total Amount   :"+Math.round(totalAmount));
				
				
		}
}

File: ui-ws/js/emi_calculator.js
google.charts.load('current', {'packages':['corechart']});
function drawChart(arr) {

    let dataArr = [[`Principal Amount`,`Interest Amount`]]
    for(let i = 0; i <arr.length;i++){
        dataArr.push(arr[i]);
    }
    let data = google.visualization.arrayToDataTable(dataArr);
    let options = {
        title: 'Emi Calculator'
    };

    let chart = new google.visualization.PieChart(document.getElementById('idPiechart'));

    chart.draw(data, options);
}
function calculateEmi(){
    var p = document.getElementById("idAmount").value;
    var t =document.getElementById("idDuration").value;
    var r = document.getElementById("idInterest").value;
    ri = r / 12 / 100;
    emi = Math.round(p * ri * Math.pow((1+ri),t) / (Math.pow((1+ri),t) - 1));
    totalAmount = emi * t;
    totalInterest = totalAmount - p;
    console.log(totalInterest);
    let str =  `<table class="table table-bordered table-striped">
                <tr>
                    <td>EMI</td>
                    <td>${emi}</td>
                </tr>
                <tr>
                    <td>Total Amount</td>
                    <td>${totalAmount}</td>
                </tr>
                <tr>
                    <td>Total Interest</td>
                    <td>${totalInterest}</td>
                </tr>
                </table>`;
    document.getElementById("idEmiDetails").innerHTML=str;
    google.charts.setOnLoadCallback(drawChart([[`Principle Amount`, Number.parseFloat(p)],[`Interest Amount`,totalInterest]]));
}
document.getElementById("idAmount").addEventListener("keyup", calculateEmi);
document.getElementById("idInterest").addEventListener("keyup", calculateEmi);
document.getElementById("idDuration").addEventListener("keyup", calculateEmi);

window.onload = function(){
    calculateEmi();
}



